cypress/e2e/admin/project_settings.cy.ts

```diff
@@ -0,0 +1,804 @@
+
+    cy.visit(this.page_uri)
+      .get_by_testid("hard-deadline-input")
+      .should("be.visible")
+      .click();
+
+    cy.get_by_testid("hard-deadline-picker")
+      .should("be.visible")
+      .find(".available-day")
+      .contains(new_date)
+      .should("be.visible")
+      .click();
+
+    // FIXME?: It feels weird that the period automatically switches to AM when typing a new value
+    // even without clicking here the period would switch after typing the new hours below
+    cy.get_by_testid("period-input")
+      .should("be.visible")
+      .should("have.value", "PM")
+      .click()
+      .should("have.value", "AM");
+
+    cy.get_by_testid("hour-input")
+      .should("be.visible")
+      .should("have.value", "12")
+      .type(new_hours);
+
+    cy.get_by_testid("minute-input")
+      .should("be.visible")
+      .should("have.value", "01")
+      .type(new_minutes);
+
+    // TZ environment variable is set within the npm script
+    cy.get_by_testid("timezone-select").should(
+      "have.value",
+      "America/Los_Angeles",
+    );
+
+    cy.get_by_testid("period-input")
+      .should("be.visible")
+      .should("have.value", "AM")
+      .click()
+      .should("have.value", "PM");
+
+    // check date has been updated on page
+    cy.get_by_testid("hard-deadline-input").should(
+      "contain.text",
+      new_datetime_str,
+    );
+
+    // save amd check that new data is loaded on refresh
+    cy.save_and_reload()
+      .get_by_testid("hard-deadline-input")
+      .should("contain.text", new_datetime_str);
+
+    // check that hard deadline can be deleted
+    cy.get_by_testid("clear-hard-deadline").should("be.visible").click();
+    cy.get_by_testid("hard-deadline-input").should(
+      "not.contain.text",
+      new_datetime_str,
+    );
+
+    cy.save_and_reload()
+      .get_by_testid("hard-deadline-input")
+      .should("not.contain.text", new_datetime_str);
+  });
+
+  it("shows deadline related API errors", function () {
+    const now = new Date(2024, 3, 14, 12, 1);
+    cy.clock(now);
+    const invalid_soft_deadline = "17";
+    const valid_soft_deadline = "15";
+    const hard_deadline = "16";
+
+    cy.visit(this.page_uri)
+      .get_by_testid("soft-deadline-input")
+      .should("be.visible")
+      .click()
+      .get_by_testid("soft-deadline-picker")
+      .should("be.visible")
+      .find(".available-day")
+      .contains(invalid_soft_deadline)
+      .should("be.visible")
+      .click()
+      .get_by_testid("hard-deadline-input")
+      .should("be.visible")
+      .click()
+      .get_by_testid("hard-deadline-picker")
+      .should("be.visible")
+      .find(".available-day")
+      .contains(hard_deadline)
+      .should("be.visible")
+      .click();
+
+    cy.save()
+      .get_api_errors()
+      .should("have.length", 1)
+      .first()
+      .should("contain.text", "closing time");
+
+    // make sure error goes away after valid request
```

Comment:
This suggests that there would be a check such as `.get_api_errors().should("have.length", 0)`, but I don't see that below. Should there be?

Reply:
Ah this isn't very clear. `save_and_reload()` verifies that there are no API errors after saving and after reloading. It's mentioned in the jsdoc but maybe I should rename the action? Or get rid of the action and just call `save().get_api_errors() ... refresh().get_api_errors() ...` ?